plugins/filter/time.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, René Moser <mail@renemoser.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import re
+from ansible.errors import AnsibleFilterError
+
+
+UNIT_FACTORS = {
+    'ms': [],
+    's': [1000],
+    'm': [1000, 60],
+    'h': [1000, 60, 60],
+    'd': [1000, 60, 60, 24],
+    'w': [1000, 60, 60, 24, 7],
+    'y': [1000, 60, 60, 24, 365],
+}
+
+
+def multiply(factors):
+    result = 1
+    for factor in factors:
+         result = result * factor
+    return result
+
+
+def divide(divisors):
+    result = 1
+    for divisor in divisors:
+         result = result / divisor
+    return result
+
+
+def to_time_unit(human_time, unit='ms'):
+    ''' Return a time unit from a human readable string '''
+    if unit not in UNIT_FACTORS:
```

Comment:
Maybe it would also be nice to allow `hour` and `hours` instead of just `h`. You could do that by adding another dict `UNIT_TO_SHORT_FORM` with enries `'hour': 'h', 'hours': 'h', 'minute': 'm', 'minutes': 'm', ...`, and doing `unit = UNIT_TO_SHORT_FORM.get(unit, unit)` before this line.

Reply:
nice idea